session/database.rst

```diff
@@ -0,0 +1,600 @@
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <services>
+                <service id="Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler" public="false">
+                    <argument>env(DATABASE_URL)</argument>
+
+                    <!-- you can also use PDO configuration, but requires passing two arguments: -->
+                    <!-- <argument>mysql:dbname=mydatabase, host=myhost</argument>
+                        <argument type="collection">
+                            <argument key="db_username">myuser</argument>
+                            <argument key="db_password">mypassword</argument>
+                        </argument> -->
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler;
+
+        $storageDefinition = $container->autowire(PdoSessionHandler::class)
+            ->setArguments([
+                '%env(DATABASE_URL)%',
+                // you can also use PDO configuration, but requires passing two arguments:
+                // 'mysql:dbname=mydatabase; host=myhost; port=myport',
+                // ['db_username' => 'myuser', 'db_password' => 'mypassword'],
+            ])
+        ;
+
+Next, use the :ref:`handler_id <config-framework-session-handler-id>`
+configuration option to tell Symfony to use this service as the session handler:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            session:
+                # ...
+                handler_id: Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
+        <framework:config>
+            <!-- ... -->
+            <framework:session
+                handler-id="Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler"/>
+        </framework:config>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        use Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler;
+
+        // ...
+        $container->loadFromExtension('framework', [
+            // ...
+            'session' => [
+                'handler_id' => PdoSessionHandler::class,
+            ],
+        ]);
+
+.. caution::
+
+    If the session data doesn't fit in the data column, it might get truncated
```

Comment:
AFAIK, only MySQL in non-strict mode is truncating things. Other engines are failing in such case.